prov/sockets/src/sock_ep.c

```diff
@@ -1732,6 +1737,8 @@ int sock_alloc_endpoint(struct fid_domain *domain, struct fi_info *info,
 		}
 		tx_ctx->ep_attr = sock_ep->attr;
 		tx_ctx->domain = sock_dom;
+		if (tx_ctx->rx_ctrl_ctx && tx_ctx->rx_ctrl_ctx->is_ctrl_ctx)
+			tx_ctx->rx_ctrl_ctx->domain = sock_dom;
```

Comment:
Blah - the current code is a mess in this area.  For now, I think it would help if the changes from this patch were squished into sock_tx_alloc_ctx().  That appears to be the only place where the domain isn't initialized. (You will need to pass the domain through to that call.)

All of this should really have proper reference counting, but that's a bigger change than I want to deal with for now.